src/huggingface_hub/_commit_api.py

```diff
@@ -572,9 +576,20 @@ def _fetch_lfs_files_to_copy(
             for src_repo_file in src_repo_files:
                 if isinstance(src_repo_file, RepoFolder):
                     raise NotImplementedError("Copying a folder is not implemented.")
-                if not src_repo_file.lfs:
-                    raise NotImplementedError("Copying a non-LFS file is not implemented")
-                files_to_copy[(src_repo_file.rfilename, src_revision)] = src_repo_file
+                if src_repo_file.lfs:
+                    files_to_copy[(src_repo_file.path, src_revision)] = src_repo_file
+                else:
+                    from .hf_file_system import HfFileSystem, HfFileSystemResolvedPath
+
+                    resolved_path = HfFileSystemResolvedPath(
+                        repo_type=repo_type,
+                        repo_id=repo_id,
+                        revision=src_revision or revision,
+                        path_in_repo=src_repo_file.path,
+                    )
+                    files_to_copy[(src_repo_file.path, src_revision)] = HfFileSystem(
+                        endpoint=endpoint, token=token
+                    ).open(resolved_path.unresolve(), "rb")
```

Comment:
I'd prefer a plain HTTP request here instead of relying on the `fsspec` overhead. This way we ensure the file is downloaded now and not when we serialize the payload for the `/commit` request. Shouldn't be problematic to read it all at once since the file is <5MB anyway.

An optimization would be to download all regular files to copy in parallel but let's keep it "out of scope" for this PR.

```suggestion
                else:
                    # TODO: (optimization) download regular files to copy concurrently
                    headers = build_hf_headers(token=token)
                    url = hf_hub_url(
                        endpoint=endpoint
                        repo_type=repo_type,
                        repo_id=repo_id,
                        revision=src_revision or revision,
                        filename=src_repo_file.path,
                    )
                    response = get_session().get(url, headers=headers)
                    hf_raise_for_status(response)
                    files_to_copy[(src_repo_file.path, src_revision)] = response.content
```

(with this suggestion `hf_hub_url` have to be imported above from `.file_download`)